Userland/Libraries/LibWeb/Animations/KeyframeEffect.cpp

```diff
@@ -869,4 +871,68 @@ void KeyframeEffect::visit_edges(Cell::Visitor& visitor)
         visitor.visit(keyframe);
 }
 
+void KeyframeEffect::invalidate()
+{
+    if (!target())
+        return;
+
+    if (pseudo_element_type().has_value()) {
+        // StyleProperties are not saved for pseudo-elements so there is nothing to patch
+        target()->invalidate_style();
+        return;
+    }
+
+    if (!is_current() && !is_in_effect()) {
+        if (!m_ended) {
+            // if an animation has ended, patching is not possible, and complete invalidation of the element's style
+            // is required to remove cascaded animated properties.
+            target()->invalidate_style();
+            m_ended = true;
+        }
+        return;
+    }
+    m_ended = false;
+
+    auto* style = target()->computed_css_values();
+    if (!style)
+        return;
+
+    auto style_before_animation_update = style->clone();
+
+    auto& document = target()->document();
+    document.style_computer().collect_animation_into(*this, *style, CSS::StyleComputer::AnimationRefresh::Yes);
+
+    // Traversal of the subtree is necessary to update the animated properties inherited from the target element.
+    target()->for_each_in_subtree_of_type<DOM::Element>([&](auto& element) {
+        auto* element_style = element.computed_css_values();
+        if (!element_style || !element.layout_node())
+            return IterationDecision::Continue;
+
+        for_each_changed_style_property(style_before_animation_update->properties(), style->properties(), [&](CSS::PropertyID property_id, RefPtr<CSS::StyleValue const> old_value, RefPtr<CSS::StyleValue const> new_value) {
+            if (element_style->maybe_null_property(property_id) == old_value)
+                element_style->set_property(property_id, *new_value);
+        });
```

Comment:
Hmm, this looks a little suspicious.
What if the descendant *didn't* inherit the value, but just *happens* to have the same value as `old_value` here?
Then we will think "aha! it must be inherited" and change the descendant incorrectly, right?

Reply:
right, I changed to check only properties that are marked as inherited, so now it should not be a problem.